akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolConductor.scala

```diff
@@ -85,9 +90,12 @@ private object PoolConductor {
       Ports(retryMerge.in(0), retrySplit.in, route.outArray.toList)
     }
 
-  private case class SwitchCommand(rc: RequestContext, slotIx: Int)
+  sealed trait SlotCommand
+  case class DispatchCommand(rc: RequestContext) extends SlotCommand
+  case class SwitchSlotCommand(cmd: DispatchCommand, slotIx: Int) extends SlotCommand
+  case class SlotShouldConnectCommand(slotIx: Int) extends SlotCommand
```

Comment:
This is confusing. Please look at my previous example where `SwitchCommand` wraps `SlotCommand`. That would simplify the router logic which just has to unwrap.